verilog/tools/syntax/verilog_syntax.cc

```diff
@@ -141,15 +144,61 @@ static int AnalyzeOneFile(absl::string_view content,
   const auto analyzer = ParseWithLanguageMode(content, filename);
   const auto lex_status = ABSL_DIE_IF_NULL(analyzer)->LexStatus();
   const auto parse_status = analyzer->ParseStatus();
+  Json::Value json;
+
+  if (absl::GetFlag(FLAGS_machine)) {
+    json = Json::objectValue;
+    json["filename"] = std::string(filename);
+  }
+
   if (!lex_status.ok() || !parse_status.ok()) {
-    const std::vector<std::string> syntax_error_messages(
-        analyzer->LinterTokenErrorMessages());
     const int error_limit = absl::GetFlag(FLAGS_error_limit);
     int error_count = 0;
-    for (const auto& message : syntax_error_messages) {
-      std::cout << message << std::endl;
-      ++error_count;
-      if (error_limit != 0 && error_count >= error_limit) break;
+    if (!absl::GetFlag(FLAGS_machine)) {
+      const std::vector<std::string> syntax_error_messages(
+          analyzer->LinterTokenErrorMessages());
+      for (const auto& message : syntax_error_messages) {
+        std::cout << message << std::endl;
+        ++error_count;
+        if (error_limit != 0 && error_count >= error_limit) break;
+      }
+    } else {
```

Comment:
Suggest factoring most of this code block to `verilog/analysis/json_diagnostics.h,cc`.

Reply:
Done.